src/installer/managed/Microsoft.NET.HostModel/AppHost/ElfUtils.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using System.IO.MemoryMappedFiles;
+using System.Runtime.CompilerServices;
+using System.Text;
+
+namespace Microsoft.NET.HostModel.AppHost
+{
+    internal static class ElfUtils
+    {
+        // The Linux Headers are copied from elf.h
+
+#pragma warning disable 0649
+        struct ElfHeader
+        {
+            byte EI_MAG0;
+            byte EI_MAG1;
+            byte EI_MAG2;
+            byte EI_MAG3;
+
+            public bool IsValid()
+            {
+                return EI_MAG0 == 0x7f &&
+                       EI_MAG1 == 0x45 &&
+                       EI_MAG2 == 0x4C &&
+                       EI_MAG3 == 0x46;
+            }
+        }
+#pragma warning restore 0649
+
+        unsafe public static bool IsElfImage(string filePath)
+        {
+            using (var mappedFile = MemoryMappedFile.CreateFromFile(filePath, FileMode.Open, mapName: null, capacity: 0, MemoryMappedFileAccess.Read))
```

Comment:
Might be better for this function to use a good ol' `read()` instead, as you're only interested in reading 4 bytes of a file. You won't probably even need this method to be `unsafe` also.

Reply:
Thanks @lpereira, I've fixed this for all three binary utils now.